### PR TITLE
fix: get Xiaohongshu fulltext

### LINKED ISSUE #16300: 小红书文章列表加载不出来
### Routes

```routes
/xiaohongshu/user/:user_id/notes/fulltext
```

### Full routes

```fullroutes
/xiaohongshu/user/6477f4590000000012034f1f/notes/fulltext
```

### Related documentation

https://docs.rsshub.app/zh/routes/social-media#unknown-3

### What is expected?

能正常加载处理

### What is actually happening?

image
image

### Deployment information

RSSHub demo (https://rsshub.app)

### Deployment information (for self-hosted)

_No response_

### Additional info

```shell
Bad gateway Error code 502
Visit cloudflare.com for more information.
2024-07-30 03:01:57 UTC
```

### This is not a duplicated issue

- [X] I have searched [existing issues](https://github.com/DIYgod/RSSHub/issues) to ensure this bug has not already been reported

### PR DESCRIPTION
**Pull request by rien7**
## Involved Issue / 该 PR 相关 Issue

Close #16300

## Example for the Proposed Route(s) / 路由地址示例

```routes
/xiaohongshu/user/5b55f1534eacab0302da1f02/notes/fulltext
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Anti-bot or rate limit / 反爬/频率限制
  - [x] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

Need to add `XIAOHONGSHU_COOKIE`

### CODE REVIEW

**Comment by zucchiniEvader** [comment]
有希望吗

**Comment by rien7** [comment]
> 有希望吗

需要自建

**Comment by zucchiniEvader** [comment]
> > 有希望吗
> 
> 需要自建

好的，我拉下你的代码试下

**Comment by dddaniel1** [comment]
> > 有希望吗
> 
> 需要自建

代码似乎还有点问题，我拉你仓库改了下，能用了。

**Comment by zucchiniEvader** [comment]
会合并吗 @DIYgod

File: lib/routes/xiaohongshu/notes.ts [hunk 1]
```diff
@@ -15,25 +18,125 @@ export const route: Route = {
...
+            link: url,
+            item: notes.map((item) => formatNote(url, item)),
+        };
+    }
+}
+
+async function getUser(url, cookie) {
+    const res = await got(url, {
+        headers: {
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36',
```
Comment (TonyRL) [inline lib/routes/xiaohongshu/notes.ts:69]: RSSHub will use a randomised user agent of [Chrome on mac](https://github.com/DIYgod/RSSHub/blob/a41cfc79ec93186793bf5b98a1c46cbd9a03edd3/lib/config.ts#L440) by default. Does the site only work with this fixed version of Chrome on Windows?

File: lib/routes/xiaohongshu/notes.ts [hunk 2]
```diff
@@ -15,25 +18,125 @@ export const route: Route = {
...
-        image: user.imageb || user.images,
-        link: url,
-        item: notes.map((item) => formatNote(url, item)),
-    };
+async function getFullNote(link) {
+    const cookie = config.xiaohongshu.cookie;
+    const data = (await cache.tryGet(link, async () => {
+        const res = await got(link, {
+            headers: {
+                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36',
```
Comment (TonyRL) [inline lib/routes/xiaohongshu/notes.ts:112]: RSSHub will use a randomised user agent of [Chrome on mac](https://github.com/DIYgod/RSSHub/blob/a41cfc79ec93186793bf5b98a1c46cbd9a03edd3/lib/config.ts#L440) by default. Does the site only works with this fixed version of Chrome on Windows?

**Pull request closed without merging**